index.js

```diff
@@ -13,9 +13,12 @@ module.exports = function(source, map) {
   var options = loaderUtils.getOptions(this) || {};
   var context = options.context || this.context || this.rootContext;
   var emitFile = !options.noEmit;
-  delete options.noEmit;
+  
+  // Make sure to not modify options object directly
+  var creatorOptions = JSON.parse(JSON.stringify(options));
```

Comment:
Cannot we use `Object.assign` here?

Reply:
You're right, no need for deep copy, fixed.